executor/executor.go

```diff
@@ -141,7 +141,9 @@ func newBaseExecutor(ctx sessionctx.Context, schema *expression.Schema, id fmt.S
 		maxChunkSize: ctx.GetSessionVars().MaxChunkSize,
 	}
 	if ctx.GetSessionVars().StmtCtx.RuntimeStatsColl != nil {
-		e.runtimeStats = e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.GetRootStats(e.id.String())
+		if e.id != nil {
+			e.runtimeStats = e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.GetRootStats(e.id.String())
```

Comment:
how about modifying `RuntimeStatsColl.GetRootStats()` to accept a `fmt.Stringer`, thus we can move this validation into that function, reduce a lot of `if` statements when we call this function outside.

Reply:
This is the only place that needs to check if id is nil.